src/SunburstVisual.ts

```diff
@@ -0,0 +1,443 @@
+            percentageLabel: createClassAndSelector("sunburst__percentage-label")
+        };
+        private colors: IColorPalette;
+        private selectionManager: ISelectionManager;
+        private tooltipService: ITooltipServiceWrapper;
+        constructor(options: VisualConstructorOptions) {
+            this.visualHost = options.host;
+            this.tooltipService = createTooltipServiceWrapper(
+                options.host.tooltipService,
+                options.element);
+
+            this.arc = d3.svg.arc<SunburstSlice>()
+                .startAngle((slice: SunburstSlice) => slice.x)
+                .endAngle((slice: SunburstSlice) => slice.x + slice.dx)
+                .innerRadius((slice: SunburstSlice) => Math.sqrt(slice.y))
+                .outerRadius((slice: SunburstSlice) => Math.sqrt(slice.y + slice.dy));
+
+            this.colors = options.host.colorPalette;
+            this.selectionManager = options.host.createSelectionManager();
+
+            this.svg = d3.select(options.element)
+                .append("svg")
+                .attr("viewBox", `0 0 ${Sunburst.ViewBoxSize}  ${Sunburst.ViewBoxSize}`)
+                .attr("width", "100%")
+                .attr("height", "100%")
+                .attr("preserveAspectRatio", "xMidYMid meet")
+                .classed(this.appCssConstants.main.class, true);
+
+            this.main = this.svg.append("g");
+            this.main.attr(CssConstants.transformProperty, translate(Sunburst.CentralPoint, Sunburst.CentralPoint));
+
+            this.selectedCategoryLabel = <d3.Selection<string>>this.svg
+                .append("text")
+                .classed(this.appCssConstants.label.class, true)
+                .classed(this.appCssConstants.categoryLabel.class, true);
+            this.selectedCategoryLabel.attr("text-anchor", "middle");
+            this.selectedCategoryLabel.attr("x", Sunburst.CentralPoint);
+            this.selectedCategoryLabel.attr("y", Sunburst.CentralPoint);
+
+            this.percentageLabel = <d3.Selection<string>>this.svg
+                .append("text")
+                .classed(this.appCssConstants.label.class, true)
+                .classed(this.appCssConstants.percentageLabel.class, true);
+            this.percentageLabel.attr("x", Sunburst.CentralPoint);
+            this.percentageLabel.attr("y", Sunburst.CentralPoint);
+            this.percentageLabel.attr("text-anchor", "middle");
+
+            this.svg.on("click", () => {
+                this.svg
+                    .classed(this.appCssConstants.mainInteractive.class, false);
+                this.labelsHidden = true;
+                this.selectionManager.clear();
+            });
+        }
+
+        public update(options: VisualUpdateOptions): void {
+            // if (options.type !== VisualUpdateType.Data && options.type !== VisualUpdateType.All) {
+            if (options.type !== 2 && options.type !== 62) {
```

Comment:
Why do you use the magic numbers here? I suppose that we might use the enum instead.